WordPress/Classes/ViewRelated/Blog/Site Management/DestructiveAlertHelper.swift

```diff
@@ -0,0 +1,56 @@
+import UIKit
+
+protocol DestructiveAlertHelperLogic {
+    var valueToConfirm: String? { get }
+    var alert: UIAlertController? { get }
+    func makeAlertWithConfirmation(title: String, message: String, valueToConfirm: String, destructiveActionTitle: String, destructiveAction: @escaping () -> Void) -> UIAlertController
+}
+
+class DestructiveAlertHelper: DestructiveAlertHelperLogic {
```

Comment:
Thank you for making this code reusable!

What do you think defining this logic in a UIAlertController extension (e.g. in `UIAlertController+Helpers`, or a new `UIAlertController+Destructive` etc)?

Reply:
Well I think it would be great to have `convenience init` inside an extension instead of `make` method of a helper class, but I think it's not doable, since we need something to encapsulate the value which needs to be confirmed (and pass it to `textFieldDidChange` in order to validate the input). So I think a custom `UIAlertController` subclass and a helper class (which I opted for) would work in this case.

Reply:
> I think it's not doable, since we need something to encapsulate the value which needs to be confirmed (and pass it to textFieldDidChange in order to validate the input). So I think a custom UIAlertController subclass and a helper class (which I opted for) would work in this case.

Ahh ok, I understand now!